Open Judge System/Services/OJS.Services/Business/ParticipantScores/ParticipantScoresBusinessService.cs

```diff
@@ -44,6 +59,146 @@ public void NormalizeAllPointsThatExceedAllowedLimit()
             }
         }
 
+        public CategoryContestsParticipationSummary GetCategoryParticipationSummary(int categoryId, bool showHidden, bool official = true)
+        {
+            var contests = this.contestsDataService
+                .GetAllNotDeletedByCategory(categoryId, showHidden)
+                .Include(c => c.ProblemGroups)
+                .Include(c => c.ProblemGroups.Select(pg => pg.Problems))
+                .OrderByDescending(c => c.CreatedOn)
+                .ToList();
+
+            var categoryResults = contests
+                .Select(c => this.GetParticipationSummaryForContest(c, official))
+                .ToList();
+            
+            var maxProblemsCount = categoryResults
+                .Select(ps => ps.ProblemsCount)
+                .Max();
+
+            return new CategoryContestsParticipationSummary
+            {
+                MaxProblemsCount = maxProblemsCount,
+                Results = categoryResults,
+            };
+        }
+        
+        public ParticipationsSummaryServiceModel GetParticipationSummaryForContest(Contest contest, bool official)
+        {
+            var participants = 
+                        participantsData.GetAllByContestAndIsOfficial(contest.Id, official)
```

Comment:
Make a method that accepts `contestIds` as a parameter and returns all participants that are in any of these contests.